src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2116,6 +2116,14 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Bind
                         {
                             Error(diagnostics, ErrorCode.ERR_InvalidPrimaryConstructorParameterReference, node, parameter);
                         }
+                        else if (parameter.ContainingSymbol is NamedTypeSymbol { IsExtension: true } &&
+                                (InParameterDefaultValue || InAttributeArgument ||
+                                 this.ContainingMember() is not { Kind: not SymbolKind.NamedType, IsStatic: false } || // We are not in an instance member
+                                 (object)this.ContainingMember().ContainingSymbol != parameter.ContainingSymbol) &&
```

Comment:
>(object)this.ContainingMember().ContainingSymbol != parameter.ContainingSymbol [](http://example.com/codeflow?start=33&length=78)

Is this condition necessary?

Reply:
That is for references i nested types. ```IsInDeclaringTypeInstanceMember``` does the same. Declaring a nested type in extension is an error, but I prefer to be consistent

Reply:
Consider adding a test for this